heterarchy.litcoffee

```diff
@@ -63,7 +63,18 @@ behaves like a class that would be have such a hierarchy.
             class Result extends generate tail linearization
                 __mro__: linearization
                 constructor: reparent next, @, next::constructor
+                # FILL UP MISSING CLASS ATTRIBUTES
+                # copy class attributes from `next` to `this` that only exist on `next`
                 copyOwn next, @
+                # ADJUST CLASS METHOD (so the MRO is used)
+                # copy class methods from `next` to `this` that are implemented by `next`
+                # note: this is basically the same as copyOwn but ignoring `if not to.hasOwnProperty key`
+                # This is due to the fact that when inheriting all class attributes are copied to `this` by CoffeeScript.
+                for own key, value of next when value instanceof Function
+                    @[key] = partial(reparent, next, @)(value)
```

Comment:
Me, too; and the tests pass. I inlined the `copyOwn` function and got rid of the extra check for `value instanceof Function` to see how close that would be to your suggestion. Doing so resulted in almost the same code as yours. The only difference is that `partial` function is used.
My main question is: Why was `partial` used in the first place? The following 2 lines should be equal, right?

```coffee
partial(reparent, next, @) value
reparent next, @, value
```
If so, using `partial` just never made if I'm not mistaken but you must have used it for _some_ reason. 😉